### ISSUE TITLE
xmlunit is using deprecated versions of hamcrest

### ISSUE THREAD

**Issue by rhamedy** [xmlunit/xmlunit#189]
The pom.xml has the following 

```
     <dependency>
        <groupId>org.hamcrest</groupId>
        <artifactId>hamcrest-library</artifactId>
        <version>1.3</version>
      </dependency>
      <dependency>
        <groupId>org.hamcrest</groupId>
        <artifactId>hamcrest-core</artifactId>
        <version>1.3</version>
      </dependency>
```
and based on mvnrepository https://mvnrepository.com/artifact/org.hamcrest/hamcrest-library 
```
A library of Hamcrest matchers - deprecated, please use "hamcrest" instead
```

Seems like the two could be replaced with https://mvnrepository.com/artifact/org.hamcrest/hamcrest-core 

I would love to make a pull request if you guys (the maintainers) are :+1:

**Comment by bodewig** [comment]
Your hamcrest-core link also says it is deprecated and https://mvnrepository.com/artifact/org.hamcrest/hamcrest is 2.x only. This means we'd have a major version update in our dependencies if we changed things.

There is a compatibility test in https://github.com/xmlunit/xmlunit/blob/master/compat-tests/hamcrest-test.sh that demonstrates XMLUnit matchers work with Hamcrest 2.x even if we stick with the 1.x version of the API. This means our users are free to exclude the 1.3 dependency and use 2.x instead at their choice. In order to not disrupt users who want to keep using 1.3 I'd prefer to not update the dependency without a good reason.

**Comment by harkue** [comment]
I notice that version 1.3 was released on Jul 10, 2012.  It's very old version contains many bugs like `NPE in IsIterableContainingInOrder`, `arrayContaining(null, null) cause NullPointerException`.

So, I suggest upgrade to ``2.x`` would be a good choice.

And version 2.x add lot of [improvement](https://github.com/hamcrest/JavaHamcrest/releases):
```
Version 2.0.0.0 Changes
Version 2.1-rc1 also includes changes previous released (but un-tagged as version 2.0.0.0). These changes include:

Upgraded to Java 7
Build with Gradle
Publish a single jar java-hamcrest-2.0.0.0.jar
Removed deprecated methods from previous release
Improve mismatch description of hasItem/hasItems
General improvements to mismatch descriptions
Several JavaDoc improvements and corrections
Deprecated several matcher factory methods of the for "isXyz"
Fix GH issue #75 - address doclint errors reported in JDK 1.8
Fix GH issue #69 - Iterable contains in order is null-safe
Fix GH issue #59 - added equalToObject() (i.e. unchecked) method
Fix GH issue #25 - arrayContaining(null, null) cause NullPointerException
Fix GH issue #36 - string matching on regular expressions
Fix GH issue #8 - isCloseTo() shows wrong delta in mismatch description
Fix GH issue #59 - add untyped version of equalTo, named equalToObject
Fix GC issue #131 - Implement IsEmptyMap, IsMapWithSize
Fix GC issue #187 - IsArray.describeMismatchSafely() should use Matcher.describeMismatch
Fix GC issue #155 - Add Matcher implementation for files
Fix GC issue #69 - fix NPE in IsIterableContainingInOrder
```

**Comment by bodewig** [comment]
If any of XMLUnit's users want to use Hamcrest 2.x, they can. They only need to add the dependency explicitly. Our tests indicate this should work without any issues.

If - on the other hand - we'd change the Hamcrest requirement then we'd force all our user's to upgrade, even if they are perfectly happy with 1.3.

Therefore I feel - unless there is a security issue involved - we should not bump dependency version numbers at all, but rather ensure we stay forward compatible. Things are different, if staying compatible becomes a burden. I fell like AssertJ may get closer to being a burden while Hamcrest has so far been a low maintenance dependency.
